### PR TITLE
Bug fix for navigator handles

### LINKED ISSUE #438: Navigator handles showing lower than expected again
nav

### PR DESCRIPTION
**Pull request by rjmcneill**
Closes #438

### CODE REVIEW

**Comment by djmiley** [comment]
@Afredon Why was it changed to `brush.empty()` originally?

File: src/assets/js/chart/nav.js [hunk 1]
```diff
@@ -88,10 +88,10 @@
                 .yDomain(yExtent);
 
             brush.on('brush', function() {
-                var width = selection.select('.plot-area').select('.extent').attr('width');
+                var brushExtent = (brush.extent()[1][0] - brush.extent()[0][0]);
 
                 // Hide the bar if the extent has no length
-                if (width > 0) {
+                if (brushExtent > 0) {
```
Comment (jleft) [inline src/assets/js/chart/nav.js]: Could simplify to:

```js
setHide(selection, brushExtent > 0);
```

File: src/assets/js/chart/nav.js [hunk 2]
```diff
@@ -107,7 +107,7 @@
                 brushHide = false;
                 setHide(selection, brushHide);
```
Comment (jleft) [inline src/assets/js/chart/nav.js]: Could simplify to:

```js
setHide(selection, false);
```

Comment (rjmcneill) [inline src/assets/js/chart/nav.js]: I had originally left it like that just to keep the code consistent across the functions, but I'll change it and resubmit :)

File: src/assets/js/chart/nav.js [hunk 3]
```diff
@@ -107,7 +107,7 @@
...
 
-                if (brush.empty()) {
+                if (brush.extent()[0][0] - brush.extent()[1][0] === 0) {
```
Comment (jleft) [inline src/assets/js/chart/nav.js]: Can we just check if they're equal?

```js
brush.extent()[0][0] === brush.extent()[1][0]
```

**Comment by Afredon** [comment]
@djmiley to simplify code by showing intent, but I misevaluated how a 2-dimensions brush is behaving, so empty() is not working in our case.

**Comment by djmiley** [comment]
Cheers @Afredon . @RMcNeillSL nice bug fix!

File: src/assets/js/chart/nav.js [hunk 4]
```diff
@@ -88,10 +88,10 @@
                 .yDomain(yExtent);
 
             brush.on('brush', function() {
-                var width = selection.select('.plot-area').select('.extent').attr('width');
+                var brushExtent = (brush.extent()[1][0] - brush.extent()[0][0]);
```
Comment (jleft) [inline src/assets/js/chart/nav.js]: Could potentially create a function which does this:

```js
function xEmpty(brush) {
    return brush.extent()[0][0] === brush.extent()[1][0];
}
```

Might make the intent more obvious?

**Comment by jleft** [comment]
:+1:

**Pull request merged**